core/src/sumeragi/main_loop.rs

```diff
@@ -933,8 +963,15 @@ pub(crate) fn run<F: FaultInjection>(
                     Some(Message::BlockSigned(BlockSigned { hash, signatures })) => {
                         trace!(block_hash=%hash, "Received block signatures");
 
-                        let valid_signatures = current_topology
-                            .filter_signatures_by_roles(&[Role::ValidatingPeer], &signatures);
+                        let valid_signatures = if current_view_change_index >= 1 {
+                            current_topology.filter_signatures_by_roles(
+                                &[Role::ValidatingPeer, Role::ObservingPeer],
+                                &signatures,
+                            )
+                        } else {
+                            current_topology
+                                .filter_signatures_by_roles(&[Role::ValidatingPeer], &signatures)
+                        };
```

Comment:
```suggestion
                        let roles = if current_view_change_index >= 1 {
                            &[Role::ValidatingPeer, Role::ObservingPeer]
                        } else {
                            &[Role::ValidatingPeer]
                        };
                        
                        let valid_signatures = current_topology.filter_signatures_by_roles(roles, &signatures);
```

consider me as a linter :), but this is what I would find more readable

Reply:
have the same thoughts

Reply:
I will change it.